sklearn/metrics/pairwise.py

```diff
@@ -1059,13 +1064,14 @@ def _parallel_pairwise(X, Y, func, n_jobs, **kwds):
     if effective_n_jobs(n_jobs) == 1:
         return func(X, Y, **kwds)
 
-    # TODO: in some cases, backend='threading' may be appropriate
-    fd = delayed(func)
-    ret = Parallel(n_jobs=n_jobs, verbose=0)(
-        fd(X, Y[s], **kwds)
+    # enforce a threading backend to prevent data communication overhead
+    fd = delayed(_dist_wrapper)
+    ret = np.empty((X.shape[0], Y.shape[0]), dtype=X.dtype)
+    Parallel(backend="threading", n_jobs=n_jobs, verbose=0)(
+        fd(func, ret, s, X, Y[s], **kwds)
```

Comment:
Interesting that it goes faster, I would have expected to access the shared array and writing to a shared array be limiting with respect to GIL.